### DISCUSSION TITLE
Rate-Mirrors at Arch Installation

### DISCUSSION THREAD

**Discussion by jaydip-c** [westandskif/rate-mirrors#44] [Q&A] [answered]
During Arch installation, I would like to use rate-mirrors instead of reflector. Is that possible? I know it's not include in the main packages, whereas, reflector is. Also, no aur helpers are included. what are the least numbers of packages required to build rate-mirrors during installation?
Thanks

**Comment by westandskif** [comment, marked as answer]
Hi @jaydip-c ! At the moment I don't have a machine to run archiso myself, but at least I checked the below in dockerized Arch.

_The below assumes `curl` is available and you are at the [Select the mirrors](https://wiki.archlinux.org/title/installation_guide#Installation) Arch installation step. It downloads binary version of the package, so it doesn't need `gcc-libs` and `cargo` to build `rate-mirrors` from source._

```bash
export RATE_MIRRORS_VERSION=${RATE_MIRRORS_VERSION:-"v0.15.1"} \
       RATE_MIRRORS_TMPDIR="$(mktemp -d tmp.XXXXXXXXXX)" \
       RATE_MIRRORS_TMPFILE="$(mktemp)" \
    && echo $RATE_MIRRORS_TMPDIR \
    && curl \
           https://github.com/westandskif/rate-mirrors/releases/download/$RATE_MIRRORS_VERSION/rate-mirrors-$RATE_MIRRORS_VERSION-x86_64-unknown-linux-musl.tar.gz \
           --retry-connrefused -s -L -o "$RATE_MIRRORS_TMPDIR/rate-mirrors-bin.tar.gz" \
    && tar xzf "${RATE_MIRRORS_TMPDIR}/rate-mirrors-bin.tar.gz" --directory $RATE_MIRRORS_TMPDIR \
    && "$RATE_MIRRORS_TMPDIR/rate-mirrors-$RATE_MIRRORS_VERSION-x86_64-unknown-linux-musl/rate_mirrors" \
           --allow-root \
           --save=$RATE_MIRRORS_TMPFILE arch \
    && mv /etc/pacman.d/mirrorlist /etc/pacman.d/mirrorlist-backup 2>/dev/null || true \
    && mv $RATE_MIRRORS_TMPFILE /etc/pacman.d/mirrorlist \
    && chmod 0644 /etc/pacman.d/mirrorlist
```

Should you try it, please let me know here. Thank you!

**Reply by jaydip-c** [reply]
Thank you very much. I shall try it soon.

Thanks.
